platform/asterix/boot/src/drivers/pmic.c

```diff
@@ -23,43 +24,27 @@ void pmic_init(void)
 	nrfx_twi_xfer_desc_t xfer = NRFX_TWI_XFER_DESC_TX(0x6b, data, sizeof(data));
 
 	err = nrfx_twi_init(&twi, &config, NULL, NULL);
-	if (err != NRFX_SUCCESS) {
-		dbgserial_print("TWI init failed: ");
-		dbgserial_print_hex(err);
-		dbgserial_newline();
-	}
+	PBL_ASSERT(err == NRFX_SUCCESS, "TWI init failed: %d", err);
 
 	nrfx_twi_enable(&twi);
 
 	data[0] = LDSW_BASE;
 	data[1] = LDSW2LDOSEL;
 	data[2] = LDSW2LDOSEL_LDO;
 	err = nrfx_twi_xfer(&twi, &xfer, 0);
-	if (err != NRFX_SUCCESS) {
-		dbgserial_print("TWI transfer failed: ");
-		dbgserial_print_hex(err);
-		dbgserial_newline();
-	}
+	PBL_ASSERT(err == NRFX_SUCCESS, "TWI transfer failed: %d", err);
```

Comment:
For HW issues like these, we should not assert -- we should `sad_watch()` (or, in this case, `return -1` or something so that `main.c` can `sad_watch()` on our behalf) so that there is a specific error code.

Reply:
Maybe asserts should sad watch? Error handling is also pretty bad in general. I could also propagate to main sad watch there.

Reply:
I think asserts basically "should never happen" -- i.e., it's ok to use assert for TWI init failing, since it either works or it doesn't.  But things that can indicate HW failures definitely should sadwatch.  I guess the solace about error handling is that there is rarely much that can be done about it, and the platform is very deterministic at that stage, except for HW failures.

Reply:
My rule is to reserve asserts for programming errors, any other error should be either handled or propagated up.